### PR TITLE
[GNA] 4D StridedSlice, Clamp, InsertIdentityLayerPass, SubstituteScaleShiftBroadCastPass fixes

### PR DESCRIPTION
**Pull request by p-rozen**
fixed: StrideSlice/Crop was broken when crop axis != dim.size() - 1 &… added support for tensor dim > 2
fixed: clamp had hard coded upper & lower boundary to -50/50
fixed: InsertIdentityLayerPass was adding not necessary identities that was leading to exception when 2 identities was added to convolution output
fixed: SubstituteScaleShiftBroadCastPass caused invalid input tensor dimension overwrite - instead of overwrite currently reshape is injected

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -1804,6 +1810,11 @@ void GNAGraphCompiler::PWLPrimitive(InferenceEngine::CNNLayerPtr layer) {
         activation_type = GNAFakeQuantizeLayer(layer).parseAsActivation();
     }
 
+    if (it->second == kActKaldiLstmClipping) {
+        auto& clamp_layer = dynamic_cast<ClampLayer&>(*layer.get());
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1814]: Wouldn't it be better to cast to pointer and check if casting was successful?

File: inference-engine/src/gna_plugin/gna_graph_tools.hpp [hunk 2]
```diff
@@ -712,4 +712,60 @@ inline void CNNNetworkRemoveLayer(CNNLayerPtr layer, bool checkDims = true) {
...
+    if (!new_prev_layer) {
+        THROW_IE_EXCEPTION << "Cannot reconnect layer new parent is NULL";
+    }
+
+    if (layer->insData.size() < 1) {
+        THROW_IE_EXCEPTION << "Cannot reconnect layer : " << layer->name
+            << " operation supports only layers with at least 1 incomming port";
+    }
+
+    if (old_prev_layer->outData.size() != 1) {
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/gna_graph_tools.hpp:742]: it would be good to have the same check for new_prev_layer

**Review by bsochack — APPROVED** [review]

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp (hunk 1 above)
Comment (p-rozen) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp]: fixed as recommended

File: inference-engine/src/gna_plugin/gna_graph_tools.hpp (hunk 2 above)
Comment (p-rozen) [reply inference-engine/src/gna_plugin/gna_graph_tools.hpp:742]: fixed as recommended

**Pull request merged**